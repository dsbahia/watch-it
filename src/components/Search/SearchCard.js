import React, { useState } from "react";
import "../../styles/searchcard.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faInfoCircle } from "@fortawesome/free-solid-svg-icons";
import MoreDetails from "../MoreDetails";
import NoImagePlaceholder from "../../images/No-Image-Placeholder.png";

function SearchCard({ title, posterpath, movieId }) {
  const [isShown, setIsShown] = useState(false);
  const movie = "movie";

  const handleClick = () => {
    setIsShown((current) => !current);
  };

  const posterCheck = () => {
    if (posterpath === null) {
      return NoImagePlaceholder;
    }
    return `https://image.tmdb.org/t/p/original/${posterpath}`;
  };

  return (
    <div className="search-card">
      <div className="poster-card">
        {" "}
        <img
          className="poster-img"
          alt={`${title} Movie poster`}
          src={posterCheck()}
        />
      </div>
      <div className="title-card">{title}</div>
      <div className="more-details">
        <button
          className="more-details-link"
          type="button"
          onClick={handleClick}
        >
          More Details
          <FontAwesomeIcon icon={faInfoCircle} className="icon" />
        </button>
        {isShown ? <MoreDetails type={movie} id={movieId} /> : null}
      </div>
    </div>
  );
}

export default SearchCard;
