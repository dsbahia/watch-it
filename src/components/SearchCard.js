import React from "react";
import "../styles/searchcard.css";

function SearchCard({ title, posterpath, overview }) {
  return (
    <div className="search-card">
      <div className="poster-card">{posterpath}</div>
      <div className="title-card">{title}</div>
      <div className="overview-card">{overview}</div>
    </div>
  );
}

export default SearchCard;
