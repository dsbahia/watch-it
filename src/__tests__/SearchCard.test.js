import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SearchCard from "../components/SearchCard";

describe("SearchCard", () => {
  const title = "Guardians of the Galaxy";
  const posterpath =
    "https://image.tmdb.org/t/p/w92//r2J02Z2OpNTctfOSN1Ydgii51I3.jpg";
  const overview = "Peter Quill, still reeling from the loss of Gamora";

  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SearchCard title={title} posterpath={posterpath} overview={overview} />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays component with provided props", () => {
    const { getByText, getByAltText } = render(
      <MemoryRouter>
        <SearchCard title={title} posterpath={posterpath} overview={overview} />
      </MemoryRouter>,
    );

    const posterImage = getByAltText(`${title} Movie poster`);

    expect(getByText(title)).toBeInTheDocument();
    expect(getByText(overview)).toBeInTheDocument();
    expect(getByAltText(`${title} Movie poster`)).toBeInTheDocument();
    expect(posterImage).toHaveAttribute(
      "src",
      `https://image.tmdb.org/t/p/original/${posterpath}`,
    );
  });
});
