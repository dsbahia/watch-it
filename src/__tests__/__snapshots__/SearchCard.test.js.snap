// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-card"
  >
    <div
      class="poster-card"
    >
       
      <img
        alt="Guardians of the Galaxy Movie poster"
        class="poster-img"
        src="https://image.tmdb.org/t/p/original/https://image.tmdb.org/t/p/w92//r2J02Z2OpNTctfOSN1Ydgii51I3.jpg"
      />
    </div>
    <div
      class="title-card"
    >
      Guardians of the Galaxy
    </div>
    <div
      class="overview-card"
    >
      Peter Quill, still reeling from the loss of Gamora
    </div>
    <div
      class="more-details"
    >
      <a
        class="more-details-link"
        href="/"
      >
        More Details
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-info icon"
          data-icon="circle-info"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
